### ISSUE TITLE
Please add a guide option to be able to set each item of play-services-X to the version that is required to avoid the typical play-services-X is being requested by several other libraries at [[15.x1,15.x2] ], but resolves to 15.x3.

### ISSUE THREAD

**Issue by Radecom** [dpa99c/cordova-android-play-services-gradle-release#22]
**I'm submitting a ...**  (check one with "x"):
- [ ] bug report
- [X ] feature request
- [ ] documentation issue

# Feature request

Hello @dpa99c , first of all, I want to congratulate you on your excellent work doing this amazing plugin. That's why my first recommendation is the option of being able to give a beer.

The current problem is that very often we find the error of play-services-X is being requested by several other libraries at [[15.x1,15.x2]], but resolves to 15.x3. Still using your plugin and put 15. + to be synchronized with the version of the other plugins continues to be this fearsome error.

When you are the creator of this excellent plugin it is noted that you are the expert in the theme of configuring and aligning the versions of google play services in a single one and thus avoid this annoying problem.

That's why I think you're the only one who could help everyone with a guide or how to set the version of each google android.gms: play-services-X that the user needs that we can establish each version for each subpart of google android.gms: play-services and align it with the other versions of the other plugins that we have installed in our application.

For example to be able to put these or other version variables:

com.google.android.gms: play-services-auth: 16.0.1
com.google.android.gms: play-services-base: 16.1.0
com.google.android.gms: play-services-identity: 16.0.0
com.google.android.gms: play-services-analytics: 16.0.8

the same for your firebase plugin, to be able to set the version:
com.google.firebase: firebase-analytics: 16.5.0
com.google.firebase: firebase-crash: 16.2.1

It should be noted that I already tried 2 things, using the 16. + the same error of play-services still appears-X is being requested by several other libraries at [[15.x1,15.x2]], but resolves to 15.x3.
Also, I already try to set a specific version in --variable PLAY_SERVICES_VERSION = 15.x2 tells me that it does not find the other elements in that version 15.x2.

Please help us with that big headache, you're the expert on the subject.

**Comment by fdambrosio** [comment]
thanks, I have a similar issue, and I'm unable to find the correct play-services versions

**Comment by dpa99c** [comment]
Currently this plugin (and it's sister plugins) only enables you to specify a library-wide version number (i.e. across all Play Services components) so it's not possible to specify different versions for different Play Services components.
However I think it should be possible to support this by enabling each component version to be specified via a plugin variable and providing defaults for those not specified.

This would take a major rework of the plugin and require a new major version, but since Google is continuing to move away from atomic library versioning to component-specific versions, I think it would be worth doing.
When I get some time over the next few days, I'll have a go at implementing a solution.

**Comment by IngAjVillalon** [comment]
I also think that it is an excellent idea, and I also join the congratulations and thanks to @dpa99c for taking the time to create this incredible plugin. In fact, my friend, if you give me some lines of development or some way to follow, I offer to help you with the improvements to this plugin.

**Comment by dpa99c** [comment]
I've just published `cordova-android-play-services-gradle-release@4.0.0` and `cordova-android-firebase-gradle-release@4.0.0` which should resolve this issue.

Using the example outlined in the original feature request above, you can now do this:

	cordova plugin add cordova-android-play-services-gradle-release\
		--variable play-services-auth=16.0.1\
		--variable play-services-base=16.1.0\
		--variable play-services-identity=16.0.0\
		--variable play-services-analytics=16.0.8

	cordova plugin add cordova-android-firebase-gradle-release\
		--variable firebase-analytics=16.5.0\
		--variable firebase-crash=16.5.0

**Comment by IngAjVillalon** [comment]
Wow boy, you are very big, congratulations for this excellent work, you deserve a beer because of our admiration you already have it.

**Comment by dpa99c** [comment]
No worries, I will close this now as I believe this issue is now resolved 😁

**Comment by UlisesCeca** [comment]
> cordova plugin add cordova-android-firebase-gradle-release\
> 	--variable firebase-analytics=16.5.0\
> 	--variable firebase-crash=16.5.0

Hello @dpa99c when I run those commands to fix the mentioned bug where some libraries require a version but it resolves to another... I get this when trying to build my app:

```
/platforms/android/app/src/main/java/com/adobe/phonegap/push/PushInstanceIDListenerService.java:18: error: method does not override or implement a method from a supertype
    @Override
```

How can it fix this? My original error was:

`The library com.google.android.gms:play-services-measurement-base is being requested by various other libraries at [[15.0.2,15.0.2]], but resolves to 15.0.4. Disable the plugin and check your dependencies tree using ./gradlew :app:dependencies.`
